compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1963,6 +1963,32 @@ pub trait PrettyPrinter<'tcx>: Printer<'tcx> + fmt::Write {
         })
     }
 
+    fn pretty_closure_signature(
```

Comment:
Should this method and the `pretty_closure_as_impl` reuse more code since their bodies are mostly identical?